File: C++/DFS/339.cpp
class Solution 
{
public:
    int depthSum(vector<NestedInteger>& nestedList, int height = 1) 
    {
        int sum = 0;
        for(auto& N : nestedList)
        {
            if(N.isInteger()) sum += (N.getInteger() * height);
            else sum  += depthSum(N.getList(), height+1);
        }
        return sum;
    }
};

File: Java/DFS/339.java
class Solution 
{
    private int getDepthSum(NestedInteger N, int height)
    {
        if(N.isInteger()) return N.getInteger() * height;
        int sum = 0;
        for(NestedInteger child : N.getList())
        {
            sum += getDepthSum(child, height+1);
        }
        return sum;
    }
    
    public int depthSum(List<NestedInteger> nestedList) 
    {
        int sum = 0;
        for(NestedInteger n : nestedList)
        {
            sum += getDepthSum(n, 1);
        }
        return sum;
    }
}